### ISSUE TITLE
API suggestion: download checksums

### ISSUE THREAD

**Issue by ACertainCoder** [JackboxUtility/JackboxUtility#426]
It is currently not possible to verify if the downloaded patches, launchers, etc. haven't been tampered with or are the up-to-date ones. If the API were to provide checksums, the Jackbox Utility could benefit in two ways:

1. Checking file integrity (security)
2. Comparing already downloaded files to the remote ones (as a follow-up to #424), which would allow it to re-install the already downloaded patches, if the user decided to switch back to English for a session and then want to use the patch again.

**Comment by AkiraArtuhaxis** [comment]
You can already check file integrity using Steam in the Settings. So if someone wants to go back to English they can do that.

**Comment by ACertainCoder** [comment]
That's not my point. It's not possible to check the integrity of the downloaded PATCH files. :wink: My implementation of #424 needs to assume that the file is fully downloaded and in-tact if it receives a 200 or 206 and the size matches or exceeds the remote file. Additionally, I have no info whether the locally stored patch still matches the one on the server, which would allow the Jackbox Utility to start extracting it directly, without having to re-download it first.

**Comment by AkiraArtuhaxis** [comment]
I suppose this is useful to those that want to play in English and other language at the same time. Would be nice to have more of a "Mod Manager" kind of thing where you could enable or disable "patches" so you could do that. But this seems like a nice workaround for that.

**Comment by ACertainCoder** [comment]
It would a workaround at first and could later be improved to manage downloaded patches based on their language and version, if desired. Though the security improvement is there in both cases. Especially since the Jackbox Utility downloads content that is later executed with full user access to the system.

**Comment by AlexisL61** [comment]
I agree, there should be some kind of verification support for server owners
